core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTest.java

```diff
@@ -0,0 +1,98 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.FuzzyQuery;
+import org.apache.lucene.search.NumericRangeQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.unit.Fuzziness;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class FuzzyQueryBuilderTest extends BaseQueryTestCase<FuzzyQueryBuilder> {
+
+    @Override
+    protected FuzzyQueryBuilder doCreateTestQueryBuilder() {
+        Tuple<String, Object> fieldAndValue = getRandomFieldNameAndValue(); 
+        FuzzyQueryBuilder query = new FuzzyQueryBuilder(fieldAndValue.v1(), fieldAndValue.v2());
+        if (randomBoolean()) {
+            query.fuzziness(randomFuzziness(query.fieldName()));
+        }
+        if (randomBoolean()) {
+            query.prefixLength(randomIntBetween(0, 10));
+        }
+        if (randomBoolean()) {
+            query.maxExpansions(randomIntBetween(1, 10));
+        }
+        if (randomBoolean()) {
+            query.transpositions(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.rewrite(getRandomRewriteMethod());
+        }
+        return query;
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(FuzzyQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        if (isNumericFieldName(queryBuilder.fieldName()) || queryBuilder.fieldName().equals(DATE_FIELD_NAME)) {
+            assertThat(query, instanceOf(NumericRangeQuery.class));
+        } else {
+            assertThat(query, instanceOf(FuzzyQuery.class));
+        }
+    }
+
+    @Test
+    public void testValidate() {
+        FuzzyQueryBuilder fuzzyQueryBuilder = new FuzzyQueryBuilder("", "text");
+        assertThat(fuzzyQueryBuilder.validate().validationErrors().size(), is(1));
+
+        fuzzyQueryBuilder = new FuzzyQueryBuilder("field", null);
+        assertThat(fuzzyQueryBuilder.validate().validationErrors().size(), is(1));
+
+        fuzzyQueryBuilder = new FuzzyQueryBuilder("field", "text");
+        assertNull(fuzzyQueryBuilder.validate());
+
+        fuzzyQueryBuilder = new FuzzyQueryBuilder(null, null);
+        assertThat(fuzzyQueryBuilder.validate().validationErrors().size(), is(2));
+    }
+    
+    @Test
+    public void testUnsupportedFuzzinessForStringType() throws IOException {
+        QueryShardContext context = createShardContext();
+        context.setAllowUnmappedFields(true);
+        
+        FuzzyQueryBuilder fuzzyQueryBuilder = new FuzzyQueryBuilder(STRING_FIELD_NAME, "text");
+        fuzzyQueryBuilder.fuzziness(Fuzziness.build(randomFrom("a string which is not auto", "3h", "200s")));
```

Comment:
this happens because the fieldType is not of the correct type when performing this fuzzy query. For string fields, we can only have fuzziness be an edit distance, not a time distance nor a String other than AUTO. So the asInt() method of Fuzziness is called and calls as a last resort Float.parseFloat (meaning maybe the user tried to type 1.0) which then returns a NumberFormatException.